galsim/phase_psf.py

```diff
@@ -0,0 +1,1442 @@
+        @param interpolant       Either an Interpolant instance or a string indicating which
+                                 interpolant should be used.  Options are 'nearest', 'sinc',
+                                 'linear', 'cubic', 'quintic', or 'lanczosN' where N should be the
+                                 integer order to use. [default: galsim.Quintic()]
+        @param aper              Aperture to use to compute PSF(s).
+        @param gsparams          An optional GSParams argument.  See the docstring for GSParams for
+                                 details. [default: None]
+
+        The following are optional keywords to use to setup the aperture if `aper` is not provided.
+
+        @param diam              Diameter in meters of aperture used to compute PSF from phases.
+        @param circular_pupil    Adopt a circular pupil? [default: True].
+        @param obscuration       Fractional linear circular obscuration of pupil. [default: 0.]
+        @param nstruts           Number of radial support struts to add to the central obscuration.
+                                 [default: 0]
+        @param strut_thick       Thickness of support struts as a fraction of pupil diameter.
+                                 [default: 0.05]
+        @param strut_angle       Angle made between the vertical and the first strut in the CCW
+                                 direction; must be an Angle instance.
+                                 [default: 0. * galsim.degrees]
+        @param oversampling      Optional oversampling factor for the InterpolatedImage. Setting
+                                 `oversampling < 1` will produce aliasing in the PSF (not good).
+                                 Usually `oversampling` should be somewhat larger than 1.  1.5 is
+                                 usually a safe choice.  [default: 1.5]
+        @param pad_factor        Additional multiple by which to zero-pad the PSF image to avoid
+                                 folding compared to what would be employed for a simple Airy.  Note
+                                 that `pad_factor` may need to be increased for stronger
+                                 aberrations, i.e., when the equivalent Zernike coefficients become
+                                 larger than order unity.  [default: 1.5]
+        """
+        # Assemble theta as an iterable over 2-tuples of Angles.
+        # 5 possible input kwargs cases for theta, theta_x, theta_y.
+        # 1) All undefined, in which case set to [(Angle(0), Angle(0))]
+        # 2) theta = tuple(Angle, Angle), in which case just need to listify
+        # 3) theta_x = Angle, theta_y = Angle, in which case need to listify and zip
+        # 4) theta_x/y = [Angle, Angle, ...], in which case need to zip
+        # 5) theta = [(Angle, Angle), (Angle, Angle), ...], in which case we're already done.
+        single = False
+        if 'theta' not in kwargs:  # Case 1, 3, or 4
```

Comment:
Normally a value of None would be allowed as equivalent to not specifying theta.  So a better flow here might be:
```
theta = kwargs.pop('theta',None)
if theta is None:
    ...
```